### PR TITLE
sdk-5418 optimize mini-framework

### PR DESCRIPTION
**Pull request by sergeyspryker**
Ticket: https://spryker.atlassian.net/browse/SDK-5418 

The idea behind these changes is to remove all the leftovers and obsolete code from the codebase and optimize it where it is possible in order to reduce the number of files in the project. so it can be easily used by PBC developers.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,68 +1,95 @@
...
+            -   name: CodeStyle Checks
+                run: composer cs-check
+
+            -   name: PHPStan Checks
+                run: composer stan
+
+            -   name: Run tests
+                run: vendor/bin/codecept run
+
+            -   name: Run Security check
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml:94]: security checks should be before the test as a less time-consuming step.

**Review by pavelmaksimov25 — APPROVED** [review]
Looks ok-ish. A small change should be made in the ci scripts. Must be QA-ed locally.

**Review by pavelmaksimov25 — DISMISSED** [review]

**Review by vol4onok — DISMISSED** [review]

**Review by VladislavStrelchenko — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,68 +1,95 @@
...
+
+            -   name: Validate propel files
+                run: |
+                    vendor/bin/console propel:schema:validate
+                    vendor/bin/console propel:schema:validate-xml-names
+
+            -   name: Validate transfer files
+                run: vendor/bin/console transfer:validate
+
+            -   name: Run Security check
```
Comment (gechetspr) [inline .github/workflows/ci.yml:85]: do we need a security check command in the minimal setup?

File: config/deploy-templates/deploy.base.template.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        applications:
+            Glue:
+                application: glue-backend
+                endpoints:
+                    '%hosts.glue%':
+                        store: GLOBAL
+services:
+    database:
+        engine: mysql
+        version: mariadb-10.5
```
Comment (gechetspr) [inline config/deploy-templates/deploy.base.template.yml:31]: 10.4 stated as a recommended version in our documentation

File: rector.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (gechetspr) [inline rector.php:1]: do we need to keep rector file under the version control

File: config/deploy-templates/deploy.base.template.yml (hunk 3 above)
Comment (sergeyspryker) [reply config/deploy-templates/deploy.base.template.yml]: done

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sergeyspryker) [reply .github/workflows/ci.yml:85]: Why not? It's a light check that helps to avoid security issues

**Review by gechetspr — APPROVED** [review]

**Review by vol4onok — APPROVED** [review]

**Pull request merged**